acceptance/tests/wan-federation/wan_federation_test.go

```diff
@@ -190,11 +209,259 @@ func TestWANFederation(t *testing.T) {
+			k8s.DeployKustomize(t, primarySecondaryNamepsaceOpts, cfg.NoCleanupOnFailure, cfg.NoCleanup, cfg.DebugDirectory, "../fixtures/cases/wan-federation/dc1-ns2-static-server")
+
+			// There is currently an issue that requires the intentions and resolvers to be created after
+			// the static-server/clients when using namespaces. When created before, Consul gives a "namespace does not exist"
+			// error
+			if c.secure {
+				// Only need to create intentions in the primary datacenter as they will be replicated to the secondary
+				// ns1 static-client (source) -> ns1 static-server (destination)
+				primaryHelper.CreateIntention(t, connhelper.IntentionOpts{DestinationNamespace: primaryNamespaceOpts.Namespace, SourceNamespace: primaryNamespaceOpts.Namespace})
+
+				// ns1 static-client (source) -> ns2 static-sever (destination)
+				primaryHelper.CreateIntention(t, connhelper.IntentionOpts{DestinationNamespace: primarySecondaryNamepsaceOpts.Namespace, SourceNamespace: primaryNamespaceOpts.Namespace})
+			}
+
+			// Create a service resolver for failover
+			logger.Log(t, "Creating service resolver")
+			k8s.DeployKustomize(t, primaryNamespaceOpts, cfg.NoCleanupOnFailure, cfg.NoCleanup, cfg.DebugDirectory, "../fixtures/cases/wan-federation/service-resolver")
+
+			// Verify that we respond with the static-server in the primary datacenter
+			logger.Log(t, "Verifying static-server in dc1 responds")
+			serviceFailoverCheck(t, primaryNamespaceOpts, localServerPort, primaryDatacenter)
+
+			// Scale down the primary datacenter static-server and see the failover
+			logger.Log(t, "Scale down dc1 static-server")
+			k8s.KubectlScale(t, primaryNamespaceOpts, staticServerDeployment, 0)
+
+			// Verify that we respond with the static-server in the secondary datacenter
+			logger.Log(t, "Verifying static-server in dc2 responds")
+			serviceFailoverCheck(t, primaryNamespaceOpts, localServerPort, secondaryDatacenter)
+
+			// scale down the primary datacenter static-server and see the failover
+			logger.Log(t, "Scale down dc2 static-server")
+			k8s.KubectlScale(t, secondaryNamespaceOtps, staticServerDeployment, 0)
+
+			// Verify that we respond with the static-server in the secondary datacenter
+			logger.Log(t, "Verifying static-server in secondary namespace (ns2) responds")
+			serviceFailoverCheck(t, primaryNamespaceOpts, localServerPort, secondaryNamespace)
+		})
+	}
+}
+
+// serviceFailoverCheck verifies that the server failed over as expected by checking that curling the `static-server`
+// using the `static-client` responds with the expected cluster name. Each static-server responds with a unique
+// name so that we can verify failover occurred as expected.
+func serviceFailoverCheck(t *testing.T, options *terratestK8s.KubectlOptions, port string, expectedName string) {
+	timer := &retry.Timer{Timeout: retryTimeout, Wait: 5 * time.Second}
+	var resp string
+	var err error
+	retry.RunWith(timer, t, func(r *retry.R) {
```

Comment:
FWIW, the locality-aware routing test I just added in `consul` had a false positive when retries were used (random LB eventually returned the correct name).

I'm pretty sure that doesn't apply here based on my understanding of how `Failover` works (all or nothing), but if that's not the case, we could do a non-retried "make a few extra calls" loop after this one to verify the host is consistent.

Again, probably not necessary here, but thought I'd mention since it was something that tripped another test up recently.

Reply:
that would happen if I wasn't mirroring the namespaces, I can try adding the second check after the retries